### DISCUSSION TITLE
Support ROCm

### DISCUSSION THREAD

**Discussion by nonetrix** [EricLBuehler/mistral.rs#321] [Ideas]
Since there is only a bug issue template I will assume it is the wrong place to ask, but ROCm support for Linux users and possibly Windows users would be really nice. Thanks for the awesome work :)

**Comment by EricLBuehler** [comment]
I think ROCm support would be great. I've been looking at [ZLUDA](https://github.com/vosen/ZLUDA), which is a CUDA -> HIP transpiler and seems pretty promising!

**Reply by nonetrix** [reply]
I personally tried it on a few things and didn't have good luck, but might have diffrent results. A another alternative would be to just use Vulkan compute, I personally think might be the way forward cross platform wise. It's about the same speed as ROCm in my experience actually but of course I imagine it is harder to implement. Could be solved by making a nice wrapper around it I imagine that just takes the same role of ROCm or CUDA instead of being more general, but I don't think anyone has done that yet really

**Comment by pstreet** [comment]
I've been working on a ROCm backend and support forked from mistral.rs and candle both recently focused primarily on getting it to work on my Strix Halo 395+ MS-S1 machine. Would love to get more feedback and folks willing to try it out. I've been putting in place a lot of features that I wanted coming across from llama.cpp as well. I don't have other hardware or GPU's to test with so I make no guarantees.

Branch to check out is ***wip/pstreet-rocm*** on these two repos:
* https://github.com/pstreet/mistral.rs
* https://github.com/pstreet/candle

You will need to install [ROCm SDK 10.0.0](https://rocm.docs.amd.com/en/latest/) as that's what I've been targeting at this time (it might work with something earlier but I haven't tested).

I'm running on Ubuntu 26.04.1 LTS (Resolute Raccoon) with Linux Kernel 7.2.6 x86_64.
